src/experiment.py

```diff
@@ -0,0 +1,188 @@
+        """
+        Prepare for multi-objective optimisation.
+
+        Args:
+            sensor_suite (SensorSuite): describes which sensors we will use for the experiment.
+            sensor_bounds (np.ndarray[float]): bounds within which a sensor can be placed.
+            depth (int, optional): how many sensors will fail at most. Defaults to 3.
+            repetitions (int, optional): number of repetitions we will average error over. Defaults to 10.
+            loss_limit (_type_, optional): maximum MSE for a successful experiment. Defaults to 80.
+        """
+        self.__sensor_suite = sensor_suite
+        num_sensors = sensor_suite.get_num_sensors()
+        self.__problem = self.__build_problem(sensor_bounds, num_sensors, 2, self.calc_MOO_loss)
+
+        self.__keys = self.__sensor_suite.calc_keys(repetitions)
+        self.__repetitions = repetitions
+        self.__loss_limit = loss_limit
+        self.__min_active = min_active
+
+
+    def __build_problem(self, sensor_bounds :np.ndarray[float], num_sensors :int, num_obj :int, loss_function :callable) -> Problem:
+        """
+        Args:
+            sensor_bounds (np.ndarray[float]): bounds within which a sensor can be placed.
+            num_sensors (int): number of sensors used.
+            num_obj (int): number of objectives to optimiser for.
+            loss_function (callable): loss function to minimse.
+
+        Returns:
+            Problem: problem object to optimise.
+        """
+        low_border = list(sensor_bounds[0]) * num_sensors
+        high_border = list(sensor_bounds[1]) * num_sensors
+        return Problem(
+            num_dim=num_sensors*self.__num_dim,
+            num_obj=num_obj,
+            loss_function=loss_function,
+            borders=[low_border, high_border]
+        )
+
+    
+    def design(self) -> any:
+        """
+        Returns:
+            any: results object containing Pareto-optimal layouts and optimisation history.
+        """
+        return self.__optimiser.optimise(self.__problem)
+
+
+    def calc_MOO_loss(self, sensor_array :np.ndarray[float]) -> list[float]:
+        """
+        Calculates the MOO loss of a specific sensor layout.
+
+        Args:
+            sensor_array (np.ndarray[float]): unshaped sensor layout from optimiser.
+
+        Returns:
+            list[float]: loss list.
+        """
+        sensor_pos = sensor_array.reshape(-1, self.__num_dim)
+        losses = np.zeros(self.__repetitions)
+        for i, key in enumerate(self.__keys):
+            num_active = np.count_nonzero(key == True)
+            if num_active >= self.__min_active:
+                self.__sensor_suite.set_active_sensors(key)
+                losses[i] = self.get_MSE(sensor_pos)
+            else:
+                losses[i] = np.argmax(losses)
+
+        expected_loss = np.mean(losses)
+        failure_chance = (losses>self.__loss_limit).sum()/self.__repetitions
```

Comment:
```suggestion
        failure_chance = (losses > self.__loss_limit).sum() / self.__repetitions
```
Missing whitespace around operators. Again, `pycodestyle`, `flake8` and/or `black` would help to capture style errors like this.